src/Globalization/Calendar.cs

```diff
@@ -41,18 +41,45 @@ static partial class CalendarExtensions
         /// week given the year, week-rule and which day of the
         /// week should be used as the first of the week.
         /// </summary>
-        
+
         public static DateTime FirstDateOfWeek(this Calendar calendar, int year, int weekOfYear, CalendarWeekRule weekRule, DayOfWeek firstDayOfWeek)
+        {
+            return FirstDateOfWeek(calendar, year, weekOfYear, weekRule, firstDayOfWeek, firstDayOfWeek);
+        }
+
+        /// <summary>
+        /// Calculates the first date that occurs in a calendar
+        /// week given the year and as defined per
+        /// <a href="https://en.wikipedia.org/wiki/ISO_week_date">ISO
+        /// week date</a>.
+        /// </summary>
+
+        public static DateTime FirstDateOfIsoWeek(this Calendar calendar, int year, int weekOfYear)
+        {
+            return calendar.FirstDateOfWeek(year, weekOfYear,
+                                            CalendarWeekRule.FirstFourDayWeek,
+                                            DayOfWeek.Monday,
+                                            DayOfWeek.Thursday);
+        }
+
+        /// <summary>
+        /// Calculates the first date that occurs in a calendar
+        /// week given the year, week-rule, which day of the
+        /// week should be used as the first of the week and which
+        /// day of the week should be used as the first of the year.
+        /// </summary>
+        
+        public static DateTime FirstDateOfWeek(this Calendar calendar, int year, int weekOfYear, CalendarWeekRule weekRule, DayOfWeek firstDayOfWeek, DayOfWeek yearFirstDayOfWeek)
         {
             // Source & credit: 
             // http://stackoverflow.com/questions/662379/calculate-date-from-week-number/914943#914943
+            // https://stackoverflow.com/a/9064954/6682
 
-            if (calendar == null) throw new ArgumentNullException("calendar");
             var jan1 = new DateTime(year, 1, 1);
-            var daysOffset = (int) firstDayOfWeek - (int) jan1.DayOfWeek;
+            var daysOffset = (int) yearFirstDayOfWeek - (int) jan1.DayOfWeek;
             var firstMonday = jan1.AddDays(daysOffset);
```

Comment:
The logic looks good to me and works as expected. The only thing is the name of the variable `firstMonday` which might be misleading, it is not always a Monday, it depends on what day was supplied as `yearFirstDayOfWeek'.